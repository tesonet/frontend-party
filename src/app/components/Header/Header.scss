$color_logout: #2f3254;
$color_logout_hover: #99cc33;
$centrify: (
  display: flex,
  justify-content: space-between,
  align-items: center
);

@mixin center_items($display, $justify-content, $align-items) {
  display: $display;
  justify-content: $justify-content;
  align-items: $align-items;
}

.header-style {
  @include center_items($centrify...);
  height: 100px;
  padding: 0 1em;
}

.logout-btn {
  @include center_items($centrify...);
  min-width: 60px;
  font-size: 13px;
  color: $color_logout;
  cursor: pointer;

  &:hover {
    color: $color_logout_hover;
    fill: $color_logout_hover;
  }
}
