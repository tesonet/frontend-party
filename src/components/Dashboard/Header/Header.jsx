import React, { Component } from "react";

import logo from "assets/images/testio.svg";
import styles from "./Header.module";

export default class DashboardHeader extends Component {
  handleLogout() {
    localStorage.removeItem("testioToken");
    this.props.props.history.push("/");
  }
  render() {
    return (
      <header className={styles.header}>
        <figure className={styles.header__logo}>
          <img src={logo} alt="Testio" width="115" height="30" />
        </figure>
        <div className={styles.header__action}>
          <button
            onClick={this.handleLogout.bind(this)}
            className={styles.header__action__logout}
          >
            Logout
          </button>
        </div>
      </header>
    );
  }
}
