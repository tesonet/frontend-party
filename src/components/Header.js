import React, { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import PropTypes from 'prop-types';
import AppLogoB from '../images/logo-b.svg';

// MUI Stuff
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Tooltip from '@material-ui/core/Tooltip';

//Icons
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

// Redux Stuff
import { connect } from 'react-redux';
import { logoutUser } from '../redux/actions/userActions';

const styles = theme => ({
  ...theme.spreadThis
});

class Header extends Component {
  handleLogout = () => {
    this.props.logoutUser();
  };

  render() {
    const { classes } = this.props;
    return (
      <div className='header'>
        <Grid
          direction='row'
          container
          justify='space-between'
          alignItems='center'
          spacing={0}
          className={classes.headerTop}
        >
          <Grid item sm>
            <img src={AppLogoB} alt='appImg' className='logoImage' />
          </Grid>
          <Grid item sm className={classes.logoutButton}>
            <Tooltip title='Logout' placement='top'>
              <Button
                onClick={this.handleLogout}
                className={classes.logoutBtn}
                startIcon={<ExitToAppIcon />}
              >
                Logout
              </Button>
            </Tooltip>
          </Grid>
        </Grid>
      </div>
    );
  }
}
const mapActionsToProps = { logoutUser };

Header.propTypes = {
  classes: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired
};

export default connect(null, mapActionsToProps)(withStyles(styles)(Header));
