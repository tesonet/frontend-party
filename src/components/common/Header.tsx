import React from 'react';
import Box from '../styled/Box';
import styled from 'styled-components';
import { ReactComponent as LogoSVG } from '../../assets/logotype-testio_black.svg';

const HeaderStyled = styled(Box)`
  height: 113px;
  display: flex;
  align-items: center;
`;

const Logo = styled(LogoSVG)`
  display: block;
  height: 30px;
  margin-left: 15px;
`;

export default function Header() {
  return (
    <HeaderStyled>
      <Logo alt="Tesonet" />
    </HeaderStyled>
  );
}
