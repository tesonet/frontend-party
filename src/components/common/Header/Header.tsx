import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import styles from './Header.module.scss';
import logo from './assets/logo-testio-small.png';
import logOut from './assets/logout.svg';
import { withRouter } from 'react-router-dom';
import { ROUTES } from '../../../routes';

interface Props extends RouteComponentProps<any> {}
const HeaderComponent: React.FC<Props> = ({ history }) => {
  const onLogOut = () => {
    localStorage.removeItem('auth-token');
    history.push(ROUTES.login);
  };
  return (
    <div className={styles.container}>
      <img data-test="header-logo" src={logo} alt="Testio" />
      <button
        data-test="logout-button"
        className={styles.logoutButton}
        onClick={onLogOut}
      >
        <img src={logOut} alt="logout" />
        Logout
      </button>
    </div>
  );
};

export const Header = withRouter(HeaderComponent);
