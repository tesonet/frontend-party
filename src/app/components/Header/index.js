import React from 'react';
import { logoDark } from '../../../assets';
import { SvgIcon } from '../';
import './Header.scss';

const Header = props => (
  <div className="header-style">
    <div>
      <img src={logoDark} alt="Testio logo" style={{ width: 110 }} />
    </div>

    <div className="logout-btn" onClick={props.logoutFunc}>
      <SvgIcon iconType="logIcon" />
      Logout
    </div>
  </div>
);

export default Header;
