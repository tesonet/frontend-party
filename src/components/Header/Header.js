import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../UI/Logo';
import IconButton from '../UI/IconButton';
import './Header.scss';

export const Header = ({ onButtonClick }) => (
  <header className="header">
    <Logo type="servers" />
    <IconButton buttonText="Logout" icon="logout" onClick={onButtonClick} />
  </header>
);

Header.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
};

export default Header;
