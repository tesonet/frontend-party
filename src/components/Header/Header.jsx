import React from "react";
import { PropTypes } from "prop-types";
import { Navbar, NavbarBrand } from "reactstrap";
import logo from "../../assets/images/logo-dark.svg";
import icon from "../../assets/icons/ico-lock.svg";
import "./Header.scss";

const Header = ({ onLogout }) => (
    <Navbar>
        <NavbarBrand>
            <img src={logo} className="header-logo" />
        </NavbarBrand>
        <a className="logout-link" onClick={onLogout}>
            <img src={icon} />
            Logout
        </a>
    </Navbar>
);

export default Header;

Header.propTypes = {
    onLogout: PropTypes.func
};
