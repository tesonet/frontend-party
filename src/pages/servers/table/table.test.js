import React from 'react';
import { shallow } from 'enzyme';
import Header from './tableHeader';
import TableRow from './tableRow';

it('table header should match snapshot', () => {
  const wrapper = shallow(
    <Header
      direction="desc"
      handleSort={() => {}}
      handleDirection={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('table header contains column names', () => {
  const wrapper = shallow(
    <Header
      direction="desc"
      handleSort={() => {}}
      handleDirection={() => {}}
    />,
  );
  expect(wrapper.contains('Server')).toEqual(true);
  expect(wrapper.contains('Distance')).toEqual(true);
});

it('table row should match snapshot', () => {
  const wrapper = shallow(
    <TableRow
      index={1}
      measurement="km"
      rowData={{ name: 'LT', distance: 10 }}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
