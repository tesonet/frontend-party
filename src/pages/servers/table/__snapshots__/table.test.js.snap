// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table header should match snapshot 1`] = `ShallowWrapper {}`;

exports[`table row should match snapshot 1`] = `ShallowWrapper {}`;
